spec/requests/saml_sessions_controller_spec.rb

```diff
@@ -221,34 +224,74 @@
 
   def raw_details_response
     {
-      providerFirmId: 22_381,
-      contactUserId: 29_562,
-      contacts: [
-        {
-          id: 568_352,
-          name: "SALLYCORNHILL",
-        },
+      firm: {
+        firmId: 3_959_183,
+        firmNumber: "50736",
+        ccmsFirmId: 10_835_831,
+        firmName: "DT SCRIPT PROVIDER 1",
+      },
+      user: {
+        userId: 454_031,
+        ccmsContactId: 0,
+        userLogin: "DT_SCRIPT_USER1",
+        name: "Dts User1",
+        emailAddress: "deepak.tanna@digital.justice.gov.uk",
+        portalStatus: "Active",
+      },
+      officeCodes: [
         {
-          id: 2_017_809,
-          name: username,
-        },
-      ],
-      providerOffices: [
-        {
-          id: 81_693,
-          name: "Test1 and Co",
+          firmOfficeId: 145_434,
+          ccmsFirmOfficeId: 34_406_595,
+          firmOfficeCode: "2Q242F",
+          officeName: "2Q242F,1 Skyscraper",
+          officeCodeAlt: "DT Script Office 1",
         },
       ],
     }
   end
 
-  def raw_404_response
+  def firm_response
     {
-      timestamp: "2020-10-27T12:15:13.639+0000",
-      status: 404,
-      error: "Not Found",
-      message: "No records found for [bob%20the%20builder]",
-      path: "/api/providerDetails/bob%20the%20builder",
+      firm: {
+        firmId: 3_959_183,
+        firmNumber: "50736",
+        ccmsFirmId: 10_835_831,
+        firmName: "DT SCRIPT PROVIDER 1",
+      },
+      offices: [
+        {
+          firmOfficeId: 145_434,
+          ccmsFirmOfficeId: 34_406_595,
+          firmOfficeCode: "2Q242F",
+          officeName: "2Q242F,1 Skyscraper",
+          officeCodeAlt: "DT Script Office 1",
+          addressLine1: "1 Skyscraper",
+          addressLine2: "1 Some Road",
+          addressLine3: "",
+          addressLine4: "",
+          city: "Metropolis",
+          county: "",
+          postCode: "LE1 1AA",
+          dxCentre: "",
+          dxNumber: "",
+          telephoneAreaCode: "01162",
+          telephoneNumber: "555124",
+          faxAreaCode: "",
+          faxNumber: "",
+          emailAddress: "me@email.uk",
+          vatRegistrationNumber: "",
+          type: "Legal Services Provider",
+          headOffice: "N/A",
+          creationDate: "2023-11-10",
+          lscRegion: "Midlands",
+          lscBidZone: "Leicester",
+          lscAreaOffice: "Nottingham",
+          cjsForceName: "Leicestershire",
+          localAuthority: "City of Leicester",
+          policeStationAreaName: "Leicester",
+          dutySolicitorAreaName: "Leicester",
+        },
+      ],
```

Comment:
As above, personally I prefer if the stubs only contain the bear minimum information to get the test to pass. This is more informative to our future selves and sets an expectation of only what we need from the API.